### PR TITLE
Issue #701 - Add config management plugins documentation

### PR DESCRIPTION
**Pull request by alexmt**

### CODE REVIEW

File: docs/application_sources.md [hunk 1]
```diff
@@ -101,3 +102,33 @@ value, in the values.yaml such that the value is stable between each comparison.
 ```
 argocd app set redis -p password=abc123
 ```
+
+## Config Management Plugins
+
+Argo CD allows integrating more config management tools using config management plugins. Following changes are required to configure new plugin:
+
+* Make sure required binaries are available in `argocd-repo-server` pod. The binaries can be added via volume mounts or using custom image (see [custom_tools](custom_tools.md)).
+* Register new plugin in `argocd-cm` ConfigMap:
```
Comment (alexec) [inline docs/application_sources.md]: "Register a new plugin"

File: docs/application_sources.md [hunk 2]
```diff
@@ -101,3 +102,33 @@ value, in the values.yaml such that the value is stable between each comparison.
...
+    - name: pluginName
+      init:                          # Optional command to initialize application source directory
+        command: ["sample command"]
+        args: ["sample args"]
+      generate:                      # Command to generate manifests YAML
+        command: ["sample command"]
+        args: ["sample args"]
+```
+
+Both `init` and `generate` commands are executed inside application source directory. Commands have access to system environment variables and following additional variables:
```
Comment (alexec) [inline docs/application_sources.md]: "inside the application source directory"

Comment (alexec) [reply docs/application_sources.md]: "Commands have access to system environment variables, including:

**Review by alexec — APPROVED** [review]

File: docs/application_sources.md (hunk 2 above)
Comment (alexmt) [reply docs/application_sources.md]: I meant to say here: all system environment variables of `argocd-repo-server` pod + additional variables. `ARGOCD_APP_NAME` and `ARGOCD_APP_NAMESPACE`.

**Review by jessesuen — APPROVED** [review]
Minor requests

File: docs/application_sources.md [hunk 3]
```diff
@@ -101,3 +102,33 @@ value, in the values.yaml such that the value is stable between each comparison.
...
+data:
+  configManagementPlugins: |
+    - name: pluginName
+      init:                          # Optional command to initialize application source directory
+        command: ["sample command"]
+        args: ["sample args"]
+      generate:                      # Command to generate manifests YAML
+        command: ["sample command"]
+        args: ["sample args"]
+```
```
[the remark this answers is not in the document]
Comment (alexmt) [reply docs/application_sources.md]: done

File: docs/application_sources.md [hunk 4]
```diff
@@ -101,3 +102,33 @@ value, in the values.yaml such that the value is stable between each comparison.
...
+* Register a new plugin in `argocd-cm` ConfigMap:
+
+```yaml
+data:
+  configManagementPlugins: |
+    - name: pluginName
+      init:                          # Optional command to initialize application source directory
+        command: ["sample command"]
+        args: ["sample args"]
+      generate:                      # Command to generate manifests YAML
```
[the remark this answers is not in the document]
Comment (alexmt) [reply docs/application_sources.md]: done

**Pull request merged**